### ISSUE TITLE
Unmet peer dependency insert-css@^0.2.0

### ISSUE THREAD

**Issue by dfcreative** [stackcss/sheetify#68]
Hi @yoshuawuyts!

`npm install sheetify`

gives

```sh
+-- UNMET PEER DEPENDENCY insert-css@^0.2.0
`-- sheetify@5.0.0
  +-- falafel@1.2.0
  | `-- isarray@0.0.1
  `-- through2@2.0.1
    `-- readable-stream@2.0.6
      `-- isarray@1.0.0
```

That would not be bad but running browserify gives `Cannot find module 'insert-css' from 'C:\Users\dmitry\Dropbox\Projects\gl-spectrum'`.

Why `inner-css` dependency is considered peer, when it is critical to run sheetify?

**Comment by ahdinosaur** [comment]
hey @dfcreative, it's a peer dependency because it's not required [if you use `css-extract`](https://github.com/stackcss/sheetify#write-to-separate-file-on-disk). that said, we should probably add some documentation that by default you should also install `insert-css`?

**Comment by dfcreative** [comment]
I see, useful! I am concerned though that 'npm install sheetify' is not enough for default use-case. Docs would be ok but I dont see why not just include thatas dep, that is a small package - overall who wins from it being a peer dep?

**Comment by yoshuawuyts** [comment]
The issue at hand here is that by using `sheetify`, `insert-css` will be required by your application. E.g. after running the transform your code turns from this:
```js
const sf = require('sheetify')
sf`.foo { color: blue; }`
```
to:
```js
const sf = 0
require('insert-css')('.foo { color: blue; }')
```
Because dependencies are recursively resolved upwards from the current project directory, a dependency of `insert-css` in the `sheetify` directory will not be found - unless in the case of `npm@3` with flat dependencies, but given that not everyone runst that, it cannot be relied upon.

Perhaps this reasoning should be added to the documentation; but as it stands now I don't see another way of guaranteeing `sheetify` to work other than relying on peer dependencies.

**Comment by dfcreative** [comment]
To `require('insert-css')('._23bb9f5e .foo { color: blue; }')` I hope though.
Now I see, thanks for the explanation, reasonable.

**Comment by kvnneff** [comment]
I just ran into this myself.  Here's how someone else is getting their peerDependencies working in both npm2 and npm3: https://codingwithspike.wordpress.com/2016/01/21/dealing-with-the-deprecation-of-peerdependencies-in-npm-3/

**Comment by yoshuawuyts** [comment]
@kvnneff ah right, that def looks odd but is probably effective - neat!

**Comment by yoshuawuyts** [comment]
Released as `v5.0.3` :sparkles:
